albumentations/augmentations/functional.py

```diff
@@ -765,6 +766,43 @@ def add_shadow(img, vertices_list):
     return image_rgb
 
 
+@ensure_contiguous
+@preserve_shape
+def add_gravel(img, gravels):
+    """Add gravel to the image.
+
+    From https://github.com/UjjwalSaxena/Automold--Road-Augmentation-Library
+
+    Args:
+        img (numpy.ndarray):
+
+    Returns:
+        numpy.ndarray:
+    """
+    non_rgb_warning(img)
+    input_dtype = img.dtype
+    needs_float = False
+
+    if input_dtype == np.float32:
+        img = from_float(img, dtype=np.dtype("uint8"))
+        needs_float = True
+    elif input_dtype not in (np.uint8, np.float32):
+        raise ValueError("Unexpected dtype {} for AddGravel augmentation".format(input_dtype))
+
+    image_hls = cv2.cvtColor(img, cv2.COLOR_RGB2HLS)
+
+    for gravel in gravels:
+        minx, maxx, miny, maxy, val = gravel
```

Comment:
mixed up x and y, because gen formula is `gravel_info = [max(y - r, 0), min(y + r, y), max(x - r, 0), min(x + r, x), r1]`

Reply:
Done